src/components/twilio-voice-dialer/public/twilio-voice-dialer.js

```diff
@@ -1 +1,51 @@
-// TODO: Implement Web Component
+const render = (x) => `
+  <div class="container">
+    <p id="status">Status: pending</p>
+    <input type="text" placeholder="recipient" id="recipient" value=${x.recipient}>
+  </div>
+`;
+
+class TwilioVoiceDialer extends HTMLElement {
+  static get observedAttributes() {
+    return ['recipient'];
+  }
+
+  device;
+  token;
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+  get recipient() {
+    return this.getAttribute('recipient');
+  }
+
+  connectedCallback() {}
+
+  disconnectedCallback() {
+    this.device.destroy();
+  }
+
+  attributeChangedCallback() {
+    this.shadowRoot.innerHTML = render(this);
+  }
+
+  async handleInit() {
+    this.device = new Twilio.Device(this.token, { logLevel: 1 });
+    this.setStatus('idle');
+  }
+
+  setStatus(status) {
+    const recipientEl = this.shadowRoot.querySelector('#recipient');
+    const statusEl = this.shadowRoot.querySelector('#status');
+    statusEl.innerText = `Status: ${status}`;
+  }
+
+  setToken(token) {
+    this.token = token;
+    this.handleInit();
```

Comment:
If the customer calls setToken multiple times, what happens? We should probably handle this scenario.

Reply:
Currently another device just gets created. There are no errors, since there is no async functionality yet. How should we handle this?

Reply:
If there's already a device instance, update the token using device.updateToken(); Also, should we handle the device life cycle in this PR? Or is that a different ticket? For example, when should we update vs destroy the device? What do we do if the component is detached, or rerendered etc?

Reply:
I was thinking this PR can handle it. I already implemented `destroy` when the element is removed from the doc: https://github.com/twilio/twilio-voice-js-reference-components/pull/2/files#diff-731372f63bb700410eff73a4a78531255e1a21231965fc053c640d4314f6dbbaR26 

If the component is rerendered, the device remains unchanged. 

Are there more usecases you think we should handle, if so I can open another ticket with the specifications there.